Install.Setup in MacOS to be compatible with CMake 3.24

Cmake 3.24 deprecated some features that are used in our install tests. When CMake 3.24 is used we get the following error in the MacOS builds:

```
nstalling: /Users/runner/work/ADIOS2/ADIOS2/macos11-xcode13_0-serial/testing/install/install/lib/cmake/adios2/adios2-config-version.cmake
-- Installing: /Users/runner/work/ADIOS2/ADIOS2/macos11-xcode13_0-serial/testing/install/install/lib/cmake/adios2/adios2-config-common.cmake
-- Installing: /Users/runner/work/ADIOS2/ADIOS2/macos11-xcode13_0-serial/testing/install/install/lib/cmake/adios2/adios2-targets.cmake
-- Installing: /Users/runner/work/ADIOS2/ADIOS2/macos11-xcode13_0-serial/testing/install/install/lib/cmake/adios2/adios2-targets-debug.cmake
-- Installing: /Users/runner/work/ADIOS2/ADIOS2/macos11-xcode13_0-serial/testing/install/install/lib/cmake/adios2/adios2-config.cmake
/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmp.36ZY9sps ~/work/ADIOS2/ADIOS2/macos11-xcode13_0-serial/Testing/install
Configuring adios2-config dummy CMake project
-- The C compiler identification is AppleClang 13.0.0.13000029
-- The CXX compiler identification is AppleClang 13.0.0.13000029
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Applications/Xcode_13.0.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Applications/Xcode_13.0.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found adios2: /Users/runner/work/ADIOS2/ADIOS2/macos11-xcode13_0-serial/testing/install/install/lib/cmake/adios2/adios2-config.cmake (found version "2.8.3") found components: C CXX 
CMake Error at CMakeLists.txt:9 (find_package):
  Found package configuration file:

    /Users/runner/work/ADIOS2/ADIOS2/macos11-xcode13_0-serial/testing/install/install/lib/cmake/adios2/adios2-config.cmake

  but it set adios2_FOUND to FALSE so package "adios2" is considered to be
  NOT FOUND.  Reason given by package:

  The following imported targets are referenced, but are missing: ffs::ffs



-- Configuring incomplete, errors occurred!
See also "/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmp.36ZY9sps/CMakeFiles/CMakeOutput.log".
See also "/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmp.36ZY9sps/CMakeFiles/CMakeError.log".
Failed
CMake Error at /Users/runner/work/ADIOS2/ADIOS2/macos11-xcode13_0-serial/cmake/install/post/cmake_install.cmake:44 (message):
  generate-adios2-config.sh returned 1, should be 0
Call Stack (most recent call first):
  /Users/runner/work/ADIOS2/ADIOS2/macos11-xcode13_0-serial/cmake_install.cmake:54 (include)


CMake Error at /Users/runner/work/ADIOS2/ADIOS2/source/testing/install/run_install.cmake:15 (message):
  Result of installation was 1, should be 0

```

We need to resolve this to keep maintaining future CMake releases.

We have temporary fixed this in https://github.com/ornladios/ADIOS2/pull/3308 by fixing the CMake used version in MacOS build to 3.23.3.

--- a/cmake/install/post/generate-adios2-config.sh.in
+++ b/cmake/install/post/generate-adios2-config.sh.in
@@ -58,8 +58,16 @@ function prefixify() {
   echo "$@" | sed "s|${PREFIX}|\${ADIOS2_PREFIX}|g"
 }
 
-PREFIX="${DESTDIR}$1"
+PREFIX="$1"
+
+if [ "${1:0:1}" != "/" ]
+then
+  # Convert relative paths to absolute based on DESTDIR
+  PREFIX="${DESTDIR:-$(pwd)}/$1"
+fi
+
 shift
+
 CMAKE="@CMAKE_COMMAND@"
 SOURCE_DIR="@ADIOS2_SOURCE_DIR@/cmake/install/post/adios2-config-dummy"
 BUILD_DIR=$(mktemp -d)
@@ -78,10 +86,10 @@ then
   export FC="@MPI_Fortran_COMPILER@"
 fi
 
-EXTRA_CMAKE_ARGS="${EXTRA_CMAKE_ARGS} -Dadios2_DIR=${PREFIX}/@CMAKE_INSTALL_CMAKEDIR@"
+export adios2_DIR="${PREFIX}/@CMAKE_INSTALL_CMAKEDIR@"
 if [ -n "@CMAKE_PREFIX_PATH@" ]
 then
-  EXTRA_CMAKE_ARGS="${EXTRA_CMAKE_ARGS} -DCMAKE_PREFIX_PATH=@CMAKE_PREFIX_PATH@"
+  export CMAKE_PREFIX_PATH="@CMAKE_PREFIX_PATH@:${CMAKE_PREFIX_PATH}"
 fi
 
 echo "Configuring adios2-config@ADIOS2_EXECUTABLE_SUFFIX@ dummy CMake project"

--- a/scripts/ci/gh-actions/macos-setup.sh
+++ b/scripts/ci/gh-actions/macos-setup.sh
@@ -19,8 +19,8 @@ sudo xcode-select --switch /Applications/Xcode_${XCODE_VER}.app
 echo "Installing CMake"
 
 {
-  readonly version="3.23.3"
-  readonly checksum="45cda7b87cad41ac407fc150e4682b85c3eb45b1977d8e89319cb3a9a6f341f3"
+  readonly version="3.24.2"
+  readonly checksum="efb11a78c064dd7c54a50b8da247254d252112c402c6e48cb7db3f9c84a4e5ad"
   readonly pkg="cmake-${version}-macos-universal.tar.gz"
   echo "${checksum}  ${pkg}" > cmake.sha256sum
 

--- a/scripts/ci/images-v2/packages.yaml.in
+++ b/scripts/ci/images-v2/packages.yaml.in
@@ -6,7 +6,7 @@ packages:
   adios2:
     variants: $(MPI_VARIANT)+blosc+bzip2+zfp+sz+png+sst+dataman+ssc+hdf5+python+fortran
   cmake:
-    version: [3.22.4]
+    version: [3.24]
   mpich:
     version: [3.3.2]
     variants: device=ch3 netmod=tcp

--- a/testing/install/Fortran/CMakeLists.txt
+++ b/testing/install/Fortran/CMakeLists.txt
@@ -4,10 +4,10 @@
 #------------------------------------------------------------------------------#
 
 cmake_minimum_required(VERSION 3.6)
-project(adios_fortran_test Fortran)
+project(adios_fortran_test Fortran C)
 enable_testing()
 
-find_package(adios2 REQUIRED)
+find_package(adios2 COMPONENTS Fortran REQUIRED)
 
 # Test using serial Fortran bindings through adios2::fortran target.
 add_executable(adios_fortran_test main_nompi_check.F90)

